src/openms/source/FORMAT/MzTab.cpp

```diff
@@ -1634,769 +1677,858 @@ namespace OpenMS
 
     // pre-analyze data for occuring meta values at feature and peptide hit level
     // these are used to build optional columns containing the meta values in internal data structures
-
     set<String> feature_user_value_keys;
     set<String> peptide_hit_user_value_keys;
+    MzTab::getFeatureMapMetaValues_(feature_map, feature_user_value_keys, peptide_hit_user_value_keys);
+
     for (Size i = 0; i < feature_map.size(); ++i)
     {
       const Feature& f = feature_map[i];
-      vector<String> keys;
-      f.getKeys(keys); //TODO: why not just return it?
-      for (String & s : keys)
-      {
-        if (s.has(' '))
-        {
-          s.substitute(' ', '_');
-        }
-      }
+      auto row = peptideSectionRowFromFeature_(f, feature_user_value_keys, peptide_hit_user_value_keys, fixed_mods);
+      mztab.getPeptideSectionRows().emplace_back(row);
```

Comment:
move row? At least for the vectors and maps inside it should work. Move for MzTabString etc might also be nice.